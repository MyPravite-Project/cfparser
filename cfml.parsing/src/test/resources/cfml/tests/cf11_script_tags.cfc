    cfhttp(url="www.google.com", method="GET");
       Cfhttp(URL="http://#CGI.SERVER_NAME#.../target.cfm", method="GET")
       {
             cfhttpparam(type="url", name='emp_name', value="Awdhesh");
             cfhttpparam(type="header", name='myheader', value="My custom header");
       }
       Writeoutput(cfhttp.filecontent); 