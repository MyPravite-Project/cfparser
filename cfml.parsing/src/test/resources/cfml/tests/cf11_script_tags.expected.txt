/*===TOKENS===*/
IDENTIFIER               <cfhttp>
'('                      <(>
IDENTIFIER               <url>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <www.google.com>
CLOSE_STRING             <">
','                      <,>
IDENTIFIER               <method>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <GET>
CLOSE_STRING             <">
')'                      <)>
';'                      <;>
IDENTIFIER               <Cfhttp>
'('                      <(>
IDENTIFIER               <URL>
'='                      <=>
HTTP                     <http://>
'#'                      <#>
IDENTIFIER               <CGI>
'.'                      <.>
IDENTIFIER               <SERVER_NAME>
'#'                      <#>
'.'                      <.>
'.'                      <.>
'.'                      <.>
'/'                      </>
IDENTIFIER               <target>
'.'                      <.>
IDENTIFIER               <cfm>
','                      <,>
IDENTIFIER               <method>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <GET>
CLOSE_STRING             <">
')'                      <)>
'{'                      <{>
IDENTIFIER               <cfhttpparam>
'('                      <(>
IDENTIFIER               <type>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <url>
CLOSE_STRING             <">
','                      <,>
IDENTIFIER               <name>
'='                      <=>
OPEN_STRING              <'>
STRING_LITERAL           <emp_name>
CLOSE_STRING             <'>
','                      <,>
IDENTIFIER               <value>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <Awdhesh>
CLOSE_STRING             <">
')'                      <)>
';'                      <;>
IDENTIFIER               <cfhttpparam>
'('                      <(>
IDENTIFIER               <type>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <header>
CLOSE_STRING             <">
','                      <,>
IDENTIFIER               <name>
'='                      <=>
OPEN_STRING              <'>
STRING_LITERAL           <myheader>
CLOSE_STRING             <'>
','                      <,>
IDENTIFIER               <value>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <My custom header>
CLOSE_STRING             <">
')'                      <)>
';'                      <;>
'}'                      <}>
IDENTIFIER               <Writeoutput>
'('                      <(>
IDENTIFIER               <cfhttp>
'.'                      <.>
IDENTIFIER               <filecontent>
')'                      <)>
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (memberExpression
                (functionCall
                  (identifierOrReservedWord (identifier cfhttp))
                  (
                  (argumentList
                    (argument
                      (identifier url)
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression (literalExpression (stringLiteral " www.google.com ")))
                            )
                          )
                        )
                      )
                    )
                    ,
                    (argument
                      (identifier method)
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression (literalExpression (stringLiteral " GET ")))
                            )
                          )
                        )
                      )
                    )
                  )
                  )
                )
              )
            )
          )
        )
      )
      ;
    )
  )
  (element statement)
  (element (statement Cfhttp))
  (element (statement ())
  (element
    (statement
      (assignmentExpression
        (startExpression
          (compareExpression
            (baseExpression (unaryExpression (memberExpression (identifier URL))))
          )
        )
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression (memberExpression (identifier (cfmlFunction http://))))
            )
          )
        )
      )
      <missing ';'>
    )
  )
  (element statement)
  (element (statement #))
  element
  (element CGI)
  (element (statement .))
  (element SERVER_NAME)
  (element (statement #))
  (element (statement .))
  (element (statement .))
  (element (statement . /))
  element
  (element target)
  (element (statement .))
  (element cfm ,)
  (element
    (statement
      (assignmentExpression
        (startExpression
          (compareExpression
            (baseExpression (unaryExpression (memberExpression (identifier method))))
          )
        )
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression
                (primaryExpression (literalExpression (stringLiteral " GET ")))
                )
              )
            )
          )
        )
      )
      <missing ';'>
    )
  )
  (element
    (statement
      (compoundStatement
        {
        (statement
          (startExpression
            (compareExpression
              (baseExpression
                (unaryExpression
                  (memberExpression
                    (functionCall
                      (identifierOrReservedWord (identifier cfhttpparam))
                      (
                      (argumentList
                        (argument
                          (identifier type)
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression (literalExpression (stringLiteral " url ")))
                                )
                              )
                            )
                          )
                        )
                        ,
                        (argument
                          (identifier name)
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression (literalExpression (stringLiteral ' emp_name ')))
                                )
                              )
                            )
                          )
                        )
                        ,
                        (argument
                          (identifier value)
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression (literalExpression (stringLiteral " Awdhesh ")))
                                )
                              )
                            )
                          )
                        )
                      )
                      )
                    )
                  )
                )
              )
            )
          )
          ;
        )
        (statement
          (startExpression
            (compareExpression
              (baseExpression
                (unaryExpression
                  (memberExpression
                    (functionCall
                      (identifierOrReservedWord (identifier cfhttpparam))
                      (
                      (argumentList
                        (argument
                          (identifier type)
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression (literalExpression (stringLiteral " header ")))
                                )
                              )
                            )
                          )
                        )
                        ,
                        (argument
                          (identifier name)
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression (literalExpression (stringLiteral ' myheader ')))
                                )
                              )
                            )
                          )
                        )
                        ,
                        (argument
                          (identifier value)
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression (literalExpression (stringLiteral " My custom header ")))
                                )
                              )
                            )
                          )
                        )
                      )
                      )
                    )
                  )
                )
              )
            )
          )
          ;
        )
        }
      )
    )
  )
  (element
    (statement
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (memberExpression
                (functionCall
                  (identifierOrReservedWord (identifier Writeoutput))
                  (
                  (argumentList
                    (argument
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (memberExpression (identifier cfhttp) . (identifier filecontent))
                            )
                          )
                        )
                      )
                    )
                  )
                  )
                )
              )
            )
          )
        )
      )
      ;
    )
  )
  (endOfScriptBlock <EOF>)
)
/*======*/